### PR TITLE
[bitnami/postgresql] Revisit tests

### PR DESCRIPTION
**Pull request by joancafom**
### Description of the change

After reviewing the existing tests and the general testing strategy we have been following until now, we saw there was a need to revisit them. The main objective that our chart testing strategy should aim for is asserting that the Chart deliverable works. As such, the focus should not be on the application itself or the container, which are regarded as trustful components (that is, they should have been tested at a previous stage).

As charts usually pack a number of different components, it is essential to test that their integrations and the application itself work as expected.

Consequently, we have updated the tests so they are related to the chart deliverable and left only a couple of basic tests to verify that the application functions correctly. Besides that, an effort has been made to reduce future tests' flakiness improving some existing selectors and checks.

Additionally, this revisit has reduced the number of executed pipelines for Redis to one, and thus its related CI file has been removed.

Tested in https://github.com/joancafom/charts/pull/88

### Checklist

- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by agomezmoron — APPROVED** [review]

File: .vib/postgresql/goss/goss.yaml [hunk 1]
```diff
@@ -1,29 +1,48 @@
...
-    exec: PGPASSWORD="$POSTGRES_PASSWORD" psql -U "$POSTGRES_USER" -d "$POSTGRES_DB"
+  {{- $dbUser := .Vars.auth.username }}
+  {{- $dbPassword := .Vars.auth.password }}
+  {{- $dbName := .Vars.auth.database }}
+  {{- $lb_port := .Vars.primary.service.ports.postgresql }}
+  {{- $port := .Vars.containerPorts.postgresql }}
+  {{- $replica_nodes := .Vars.readReplicas.replicaCount }}
+  {{- $table := printf "test_%s" (randAlpha 5) }}
+  postgresql-check-replication:
+    exec: export PGPASSWORD='{{ $dbPassword }}' && psql -U '{{ $dbUser }}' -d '{{ $dbName }}' -p {{ $lb_port }} -h postgresql-{{ .Vars.primary.name }} -c "CREATE TABLE {{ $table }} (id int, first_name VARCHAR(50));" && sleep 1 {{ range $e, $i := until $replica_nodes }} && psql -U '{{ $dbUser }}' -d '{{ $dbName }}' -p {{ $port }} -h postgresql-{{ $.Vars.readReplicas.name }}-{{ $i }}.postgresql-{{ $.Vars.readReplicas.name }}-hl -c "SELECT * FROM {{ $table }};"{{ end }}
```
Comment (FraPazGal) [inline .vib/postgresql/goss/goss.yaml:10]: Could we use a variable for `postgresql-{{ $.Vars.readReplicas.name }}`? It seems to be repeated several times in the first two tests.

File: .vib/postgresql/vib-publish.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
...
...ogIGVuYWJsZVBvc3RncmVzVXNlcjogdHJ1ZQogIHBvc3RncmVzUGFzc3dvcmQ6ICJDb21wbGljYXRlZFBhc3N3b3JkMTIzITQiCiAgdXNlcm5hbWU6IHRlc3RfdXNlcgogIHBhc3N3b3JkOiB0ZXN0X3Bhc3N3b3JkCiAgZGF0YWJhc2U6IHRlc3RfZGF0YWJhc2UKICByZXBsaWNhdGlvblVzZXJuYW1lOiB0ZXN0X3JlcGxfdXNlcgpjb250YWluZXJQb3J0czoKICBwb3N0Z3Jlc3FsOiA1NDMzCnBvc3RncmVzcWxTaGFyZWRQcmVsb2FkTGlicmFyaWVzOiAicGdhdWRpdCwgcGdfc3RhdF9zdGF0ZW1lbnRzIgpzaG1Wb2x1bWU6CiAgZW5hYmxlZDogdHJ1ZQpwcmltYXJ5OgogIG5hbWU6IHB0ZXN0CiAgcG9kU2VjdXJpdHlDb250ZXh0OgogICAgZW5hYmxlZDogdHJ1ZQogICAgZnNHcm91cDogMTAwMgogIGNvbnRhaW5lclNlY3VyaXR5Q29udGV4dDoKICAgIGVuYWJsZWQ6IHRydWUKICAgIHJ1bkFzVXNlcjogMTAwMgogIHNlcnZpY2U6CiAgICB0eXBlOiBMb2FkQmFsYW5jZXIKICAgIHBvcnRzOgogICAgICBwb3N0Z3Jlc3FsOiA1NDMzCiAgcGVyc2lzdGVuY2U6CiAgICBlbmFibGVkOiB0cnVlCiAgICBtb3VudFBhdGg6IC9iaXRuYW1pL3Bvc3RncmVzcWwKcmVhZFJlcGxpY2FzOgogIG5hbWU6IHJ0ZXN0CiAgcmVwbGljYUNvdW50OiAyCiAgcG9kU2VjdXJpdHlDb250ZXh0OgogICAgZW5hYmxlZDogdHJ1ZQogICAgZnNHcm91cDogMTAwMgogIGNvbnRhaW5lclNlY3VyaXR5Q29udGV4dDoKICAgIGVuYWJsZWQ6IHRydWUKICAgIHJ1bkFzVXNlcjogMTAwMgogIHNlcnZpY2U6CiAgICB0eXBlOiBMb2FkQmFsYW5jZXIKICAgIHBvcnRzOgogICAgICBwb3N0Z3Jlc3FsOiA4MAogIHBlcnNpc3RlbmNlOgogICAgZW5hYmxlZDogdHJ1ZQogICAgbW91bnRQYXRoOiAvYml0bmFtaS9wb3N0Z3Jlc3Fs",
```
Comment (FraPazGal) [inline .vib/postgresql/vib-publish.json:25]: No strong thoughts on this, but is it necessary to have `readReplicas.persistence.mountPath` defined here? We are maintaining its default value and changes to the persistence conf shouldn't affect our testing environment.

**Review by FraPazGal — COMMENTED** [review]
Thanks for the PR @joancafom! I left a couple of minor comments on it

File: .vib/postgresql/vib-publish.json (hunk 2 above)
Comment (agomezmoron) [reply .vib/postgresql/vib-publish.json:25]: I think @joancafom followed the rule of _all we can be customized works_

**Review by agomezmoron — APPROVED** [review]

File: .vib/postgresql/vib-publish.json (hunk 2 above)
Comment (joancafom) [reply .vib/postgresql/vib-publish.json]: Yes, that was the rule I followed. However, we discussed it internally and I changed it to only include things that we can either assert in the tests to be working or that directly affect the test environment

**Review by FraPazGal — APPROVED** [review]

**Pull request merged**